src/compass/s1_geo_stack.py

```diff
@@ -0,0 +1,468 @@
+                             ' (default: co-pol).')
+    parser.add_argument('-x', '--x-spac', type=float, default=5,
+                        help='Spacing in meters of geocoded CSLC along X-direction.')
+    parser.add_argument('-y', '--y-spac', type=float, default=10,
+                        help='Spacing in meters of geocoded CSLC along Y-direction.')
+    parser.add_argument('--bbox', nargs=4, type=float, default=None, 
+                        metavar=('lonmin', 'latmin', 'lonmax', 'latmax'),
+                        help='(Optional) Bounding box of the geocoded stack'
+                             ' in lon/lat degrees. ')
+    parser.add_argument('-e', '--epsg', type=int, default=None,
+                        help='EPSG projection code for output geocoded bursts')
+    parser.add_argument('-f', '--flatten', type=bool, default=True,
+                        help='If True, enables flattening (default: True)')
+    parser.add_argument('-ss', '--range-split-spectrum',
+                        dest='is_split_spectrum', type=bool, default=False,
+                        help='If True, enables split-spectrum (default: False)')
+    parser.add_argument('-lb', '--low-band', type=float, default=0.0,
+                        help='Low sub-band bandwidth in Hz (default: 0.0)')
+    parser.add_argument('-hb', '--high-band', type=float, default=0.0,
+                        help='High sub-band bandwidth in Hz (default: 0.0')
+    return parser.parse_args()
+
+
+def generate_burst_map(zip_files, orbit_dir, x_spac, y_spac, epsg=4326, bbox=None):
+    '''
+    Generates a dataframe of geogrid infos for each burst ID in `zip_files`.
+
+    Parameters
+    ----------
+    zip_files: str
+        List of S1-A/B SAFE (zip) files
+    orbit_dir: str
+        Directory containing sensor orbit ephemerides
+    x_spac: float
+        Spacing of geocoded burst along X-direction
+    y_spac: float
+        Spacing of geocoded burst along Y-direction
+    epsg: int
+        EPSG code identifying output product projection system
+    bbox: Optional[tuple[float]]
+        Bounding box of the geocoded bursts (left, bottom, right, top)
+        If not provided, the bounding box is computed for each burst
+
+    Returns
+    -------
+    burst_map: pandas.Dataframe
+        Pandas dataframe containing geogrid info (e.g. top-left, bottom-right
+        x and y coordinates) for each burst to process
+    '''
+    # Initialize dictionary that contains all the info for geocoding
+    burst_map = defaultdict(list)
+
+    # Get all the bursts from safe file
+    i_subswath = [1, 2, 3]
+
+    for zip_file in zip_files:
+        orbit_path = get_orbit_file_from_dir(zip_file, orbit_dir)
+
+        for subswath in i_subswath:
+            ref_bursts = load_bursts(zip_file, orbit_path, subswath)
+            for burst in ref_bursts:
+                if epsg is None:
+                    epsg = get_point_epsg(burst.center.y,
```

Comment:
I've added the use of the database instead of calculating it.

One note- I didn't know whether to have the default behavior be

1. every burst gets the EPSG from the database. So if a frame straddles a boundary, it will have some from each zone.
2. pick the first EPSG we see (which was how it was implemented), then translate the boundaries from the database to match.

I've opted for 1 for now, since you can specify one EPSG with the command line arg, but you can't easily say "use the database for everything"